HPXMLtoOpenStudio/resources/model.rb

```diff
@@ -779,21 +779,38 @@ def self.merge_unit_models(model, hpxml_osm_map)
       end
     end
 
-    hpxml_osm_map.values.each_with_index do |unit_model, unit_number|
+    unit_surface_to_obj_index_map = {} # map of unit model surface handle to whole building model object index
```

Comment:
@shorowit Sadly that there's not a better solution to keep the surface-to-surface mapping information while knowing the handles after merging unit models (and I cannot do it in front, by assigning a surface in another model space as adjacent surface). The only solution I can only think of here is to use the indexes after calling model.addObjects. OS source codes specifically call out that the order will be kept, so it's fine for now, but we need to be careful if this is called more than once in the future.